api/deployer.go

```diff
@@ -423,8 +423,12 @@ func (d *deployerSpec) monitorRollout() error {
 		if err != nil {
 			return err
 		}
+		fromRunningCount := 0
+		if d.fromReplicaSet != nil {
+			_, fromRunningCount, _ = d.refreshReplicaSetPodCount(d.fromReplicaSet)
+		}
 		log.Debugf("Replica Count: Ready: %v, Running: %v", readyCount, runningCount)
-		if int32(readyCount) == *d.toReplicaSet.Spec.Replicas {
+		if readyCount == d.deployment.DesiredReplicas && fromRunningCount == 0 {
```

Comment:
Nevermind, I could just read the code and see this is in a for loop that handles this 😁